debug/highlightpoints.html

```diff
@@ -49,16 +49,21 @@
         }
     });
     let hoveredFeature;
-    map.on('mousemove', 'places', function(e) {
-        if (e.features.length) {
-            const f = e.features[0];
+    map.on('mousemove', function(e) {
+        var features = map.queryRenderedFeatures(e.point, {layers:['places']});
+
+        if (features[0]) {
+            const f = features[0];
             if (!f.state.hover) {
-                map.setFeatureState(f, {'hover': true});
-                if (hoveredFeature) {
-                    map.setFeatureState(hoveredFeature, {'hover': false});
-                }
+                map.setFeatureState(f, {'hover': true, 'visited': true, 'visits':f.state.visits ? f.state.visits + 1 : 3});
                 hoveredFeature = f;
+
             }
+        } else if (hoveredFeature) {
```

Comment:
Can we tweak the demo to _always_ remove the previous hover state, even if a new item is being hovered (assuming it's not the same item)? I know it's not critical for actually testing the functionality, but it's helpful for future developers if the "meaning" of the demos is easy to figure out by playing with them.